internal/pkg/manifest/transform.go

```diff
@@ -18,58 +21,284 @@ var basicKinds = []reflect.Kind{
 	reflect.Complex64, reflect.Complex128,
 	reflect.Array, reflect.String, reflect.Slice,
 }
+var fmtExclusiveFieldsSpecifiedTogether = "invalid manifest: %s %s mutually exclusive with %s and shouldn't be specified at the same time"
 
 type workloadTransformer struct{}
 
 // Transformer returns custom merge logic for workload's fields.
 func (t workloadTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
```

Comment:
Instead of having a single `workloadTransformer` type that handles all possible fields in the manifest, is it possible to get to a solution that pushes as much merging logic as possible to mergo? without us doing the work.

For example is this possible:
```go
  // in transform.go
  // Special merge transformers.
  // We have to write one for each composite type provided by the manifest.
 var transformers = []mergo.Transfomers {
    imageTransfomer{},
    volumeTransfomer{},
    ...
  }

func merge(dst, src interface{}) error {
  for _, t := range transformers {
    if err := mergo.Merge(dst, src, mergo.WithOverride, mergo.WithTransfomers(t)) {
       return err
    }
  }
}
```
then in apply env we do:
```go
err := merge(&s, LoadBalancedWebService{
		LoadBalancedWebServiceConfig: *overrideConfig,
	})
```

Reply:
done!